### ISSUE TITLE
No eligible files found error when the only file to check is also ignored

### ISSUE THREAD

**Issue by mgrebenets** [nicklockwood/SwiftFormat#368]
This is an edge case that happens for our setup.
We have configured SwiftFormat to ignore any file that has `AutoGenerated` in the name.

```text
# .swiftformat
--exclude **/*AutoGenerated*
```

But if I then tell SwiftFormat to check ignored file, it gives me an error:

```bash
swiftformat path/to/AutoGenerated.swift
running swiftformat...
error: no eligible files found at /Users/user/project/path/to/AutoGenerated.swift
```

I think this doesn't look right.
If I ask SwiftFormat to format ignored file, it should just ignore it and finish with successful result.

We have this error happening in CI setup when we use SwiftFormat Danger plugin.
The danger plugin just takes all Swift files changed in pull request and feeds them to SwiftFormat.
So the Danger check fails if the only files changed are those that should be ignored.

**Comment by nicklockwood** [comment]
@mgrebenets that makes sense. This error was intended to ensure that you didn't accidentally try to format the wrong directory and then assume it had worked because there were no files changed, but I agree it's overly aggressive in this case. It should probably treat ignored files differently from no files.

**Comment by nicklockwood** [comment]
@mgrebenets this is fixed 0.39.4
